lib/DXIL/DxilOperations.cpp

```diff
@@ -3448,10 +3472,17 @@ void OP::GetMinShaderModelAndMask(const llvm::CallInst *CI,
         minor = 8;
       }
     }
+    if (BarrierRequiresReorder(CI)) {
+      major = 6;
+      minor = 9;
+      mask &= SFLAG(Library) | SFLAG(RayGeneration);
+      return;
+    }
     if (BarrierRequiresNode(CI)) {
       mask &= SFLAG(Library) | SFLAG(Node);
       return;
-    } else if (BarrierRequiresGroup(CI)) {
+    }
+    if (BarrierRequiresGroup(CI)) {
```

Comment:
I think this was `else if` just because the prior branch results in more restrictive flags, so it was unnecessary to check group if it already required node.  Not that I think it requires changing back necessarily.

Reply:
This was an else-after-return, so i cleaned it up: https://llvm.org/docs/CodingStandards.html